gobblin-compaction/src/main/java/gobblin/compaction/verify/SimulatedDataCompletenessVerifier.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.verify;
+
+import java.util.List;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import gobblin.compaction.Dataset;
+import gobblin.compaction.verify.DataCompletenessVerifier.Results;
+import gobblin.compaction.verify.DataCompletenessVerifier.Results.Result;
+import gobblin.configuration.State;
+
+
+/**
+ * A class that simulates data completeness verification for Gobblin compaction. It randomly decides
+ * whether the verification of a {@link Dataset} passes or fails.
```

Comment:
What is the purpose of the class? Is it for testing? If yes, In general, having test classes rely on random behavior is not ideal as they make reproducing errors hard. At the very least, there should be enough audit trail about all the random decisions made.